prow/tide/README.md

```diff
@@ -0,0 +1,128 @@
+components for managing a pool of PRs that match a given set of criteria.
+Automatically retesting PRs in the pool and merging them if they pass tests.
+
+Configuration of Tide is located under
+[prow/config.yaml](https://github.com/kubernetes/test-infra/blob/master/prow/config.yaml)
+file. It consists of two parts:
+
+  * **Specification of PR pools/criteria**:
+    A set of all PRs Tide keeps track of.
+    The PRs are periodically checked if they are ready to be merged.
+    Configured under `tide` key of the configuration file.
+
+  * **Specification of PR tests**:
+    A set of jobs that are run over each PR before it gets merged.
+    No job is required to be provided in which case it's enough a PR has all
+    relevant labels set. If a set of jobs is specified,
+    all must succeed (unless configured otherwise).
+    Configured under `presubmits` key of the configuration file.
+
+## PR pools
+
+The set of criteria is specified through the following collection of items:
+
+```yaml
+tide:
+  queries:                // List of queries
+  - repo:                   // org/repo
+    labels:                 // List of must have labels
+    missingLabels:          // List of can't have labels
+    excludedBranches:       // Ignore branches
+    includedBranches:       // Include-only branches
+    reviewApprovedRequired: // Review approved is must
+  sync_period:            // Sync jobs period
+  status_update_period:   // PR status update period
+  merge_method:           // Squash, rebase or just merge PR
+  target_url:             // ???
+  pr_status_base_url:     // PR status page
+  max_goroutines:         // Per-pool parallelism
+```
+
+Depending on your criteria, some of the items may by omitted.
+
+### General configuration
+
+The following configuration fields are available:
+
+* `sync_period`: The field specifies how often Tide will sync jobs with Github. Defaults to 1m.
+* `status_update_period`: The field specifies how often Tide will update Github status contexts.
+   Defaults to the value of `sync_period`.
+* `queries`: List of queries (described below).
+* `merge_method`: A key/value pair of an `org/repo` as the key and merge method to override
+   the default method of merge as value. Valid options are `squash`, `rebase`, and `merge`.
+   Defaults to `merge`.
+* `target_url`: URL for tide status contexts.
+* `pr_status_base_url`: The base URL for the PR status page.
+* `max_goroutines`: The maximum number of goroutines spawned inside the component to
+   handle org/repo:branch pools. Defaults to 20. Needs to be a positive number.
+
+### Queries
+
+The `queries` field specifies a list of queries.
+Each query corresponds to a set of **open** PRs as candidates for merging.
+It can consist of the following dictionary of fields:
+
+* `repos`: List of queried repositories.
+* `labels`: List of labels any given PR from the `repos` must posses.
+* `missingLabels`: List of labels any given PR from the `repos` must not posses.
```

Comment:
Drop the mention of `repos` from this line and above. It isn't quite correct because we are actually considering PRs from the `repos` **that are in an appropriate branch** as determined by the branch parameters.

It should be obvious enough that a PR must satisfy all the requirements in this list to be part of the query output.

Reply:
Done